src/control/lib/spdk/nvme.go

```diff
@@ -221,19 +271,18 @@ func c2GoNamespace(ns *C.struct_ns_t) *Namespace {
 }
 
 // processReturn parses return structs
-func processReturn(retPtr *C.struct_ret_t, failLocation string) (ctrlrs []Controller, err error) {
+func processReturn(retPtr *C.struct_ret_t, failMsg string) (ctrlrs []Controller, err error) {
 	if retPtr == nil {
-		return nil, errors.New("empty return value")
+		return nil, errors.Wrap(FaultBindingRetNull, failMsg)
 	}
+	ctrlrPtr := retPtr.ctrlrs
 
 	if retPtr.rc != 0 {
-		cleanReturn(retPtr)
-
-		return nil, errors.Errorf("%s failed, rc: %d, %s",
-			failLocation, retPtr.rc, C.GoString(&retPtr.err[0]))
+		err = errors.Wrap(FaultBindingNonZeroRC(int(retPtr.rc), C.GoString(&retPtr.err[0])),
+			failMsg)
+		goto clean
```

Comment:
This is much harder to understand. It's not C; we don't have to do things this way.

Instead, why not just call cleanReturn() and return early?